### ISSUE TITLE
pandas2ri - requires pytz

### ISSUE THREAD

**Issue by dyrvte** [rpy2/rpy2#1079]
Since ZoneInfo is now being used as the standard for Python I switched my pandas dataset from using pytz for timezones to ZoneInfo. It appears pandas2ri does not support this currently:

pandas2ri.py:65: UserWarning: Error while trying to convert the column ... Fall back to string conversion. The error is: 'zoneinfo.ZoneInfo' object has no attribute 'zone'

This attribute is not present in ZoneInfo

**Comment by lgautier** [comment]
@dyrvte - Can you share a minimal code example to reproduce the issue?

**Comment by dyrvte** [comment]
import rpy2.robjects as ro
from rpy2.robjects import pandas2ri as p2ri
from rpy2.robjects.conversion import localconverter
import pandas as pd
import pytz
from zoneinfo import ZoneInfo
from datetime import datetime
#works fine

record_for_prediction = pd.DataFrame([[datetime.now(),1]],columns =
["date","number"])

print(record_for_prediction)
context = localconverter(ro.default_converter + p2ri.converter)
with context as cv:
    to_predict = cv.py2rpy(record_for_prediction )
#works fine
tz = pytz.UTC

record_for_prediction = pd.DataFrame([[datetime.now(tz),1]],columns =
["date","number"])

print(record_for_prediction)
context = localconverter(ro.default_converter + p2ri.converter)
with context as cv:
    to_predict = cv.py2rpy(record_for_prediction )

#error occurs with this one
tz = ZoneInfo("UTC")

record_for_prediction = pd.DataFrame([[datetime.now(tz),1]],columns =
["date","number"])

print(record_for_prediction)
context = localconverter(ro.default_converter + p2ri.converter)
with context as cv:
    to_predict = cv.py2rpy(record_for_prediction )

**Comment by lgautier** [comment]
Thanks I can't reproduce on this end with rpy2 3.5.15, and pandas 2.1.3. Can you share the output of
```bash
python -m rpy2.situation
```
and
```
python -c 'import pandas; print(pandas.__version__)'
```

**Comment by dyrvte** [comment]
Testing with 3.5.14 and 3.5.15 on my system gives the same error.
Pandas is 2.1.4, which I've just upgraded.

rpy2 version:
3.5.15
Python version:
3.11.6 (main, Nov 14 2023, 09:36:21) [GCC 13.2.1 20230801]
Looking for R's HOME:
    Environment variable R_HOME: None
    Calling `R RHOME`: /usr/lib64/R
    Environment variable R_LIBS_USER: None
R's value for LD_LIBRARY_PATH:
/usr/lib64/R/lib:/usr/lib/jvm/java-21-openjdk/lib/server
R version:
    In the PATH: R version 4.3.2 (2023-10-31) -- "Eye Holes"
    Loading R library from rpy2: OK
Additional directories to load R packages from:
None
C extension compilation:
  include:
  ['/usr/include/R/']
  libraries:
  ['R', 'pcre2-8', 'lzma', 'bz2', 'z', 'tirpc', 'rt', 'dl', 'm',
'icuuc', 'icui18n']
  library_dirs:
  ['/usr/lib64/R/lib']
  extra_compile_args:
  ['-std=c99']
  extra_link_args:
  ['-Wl,--export-dynamic', '-fopenmp',
'-Wl,-O1,--sort-common,--as-needed,-z,relro,-z,now', '-flto=auto']
Directory for the R shared library:
lib
CFFI extension type
  Environment variable: RPY2_CFFI_MODE
  Value: CFFI_MODE.ANY
  ABI: PRESENT
  API: PRESENT

python -c 'import pandas; print(pandas.__version__)'

2.1.4

**Comment by lgautier** [comment]
Also working with pandas 2.1.4 on this end. I have Python 3.10.
If `zoneinfo` is involved it should be the same since it is available since Python 3.9 (https://docs.python.org/3/library/zoneinfo.html). I'll try with Python 3.11 later.

**Comment by lgautier** [comment]
It took me a while to get to it, but finally fixed with MR #1136. The fix will be included in rpy2-3.6.0.
